cpp/src/strings/char_types/char_types.cu

```diff
@@ -79,6 +81,102 @@ std::unique_ptr<column> all_characters_of_type( strings_column_view const& strin
     return results;
 }
 
+std::unique_ptr<column> is_integer( strings_column_view const& strings,
+                                    rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
+                                    cudaStream_t stream = 0)
+{
+    auto strings_column = column_device_view::create(strings.parent(),stream);
+    auto d_column = *strings_column;
+    // create output column
+    auto results = make_numeric_column( data_type{BOOL8}, strings.size(),
+        copy_bitmask(strings.parent(),stream,mr), strings.null_count(), stream, mr);
+    auto d_results = results->mutable_view().data<bool>();
+    thrust::transform(rmm::exec_policy(stream)->on(stream),
+        thrust::make_counting_iterator<size_type>(0),
+        thrust::make_counting_iterator<size_type>(strings.size()),
+        d_results,
+        [d_column] __device__(size_type idx){
+            if( d_column.is_null(idx) )
+                return false;
+            auto d_str = d_column.element<string_view>(idx);
+            if( d_str.empty() )
+                return false;
+            auto begin = d_str.begin();
+            if( *begin=='+' || *begin=='-' )
+                ++begin;
+            return thrust::all_of( thrust::seq, begin, d_str.end(),
+                [] __device__ (auto chr) { return chr >='0' && chr <= '9'; });
+        });
+    //
+    results->set_null_count(strings.null_count());
+    return results;
+}
+
+std::unique_ptr<column> is_float( strings_column_view const& strings,
+                                  rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
+                                  cudaStream_t stream = 0)
+{
+    auto strings_column = column_device_view::create(strings.parent(),stream);
+    auto d_column = *strings_column;
+    // create output column
+    auto results = make_numeric_column( data_type{BOOL8}, strings.size(),
+        copy_bitmask(strings.parent(),stream,mr), strings.null_count(), stream, mr);
+    auto d_results = results->mutable_view().data<bool>();
+    // check strings for valid float chars
+    thrust::transform(rmm::exec_policy(stream)->on(stream),
+        thrust::make_counting_iterator<size_type>(0),
+        thrust::make_counting_iterator<size_type>(strings.size()),
+        d_results,
+        [d_column] __device__(size_type idx){
+            if( d_column.is_null(idx) )
+                return false;
+            auto d_str = d_column.element<string_view>(idx);
+            if( d_str.empty() )
```

Comment:
It seems like it'd be better to factor this logic out into an `is_float` free function that operates on `string_view`:
```
bool is_float(string_view s){...}
```